### PR TITLE
replace group content version with content version

### PR DESCRIPTION
**Pull request by kaeline**

### CODE REVIEW

File: src/main/kotlin/fi/metatavu/noheva/api/ExhibitionDevicesGroupsApiImpl.kt [hunk 1]
```diff
@@ -89,7 +89,8 @@ class ExhibitionDevicesGroupsApiImpl : ExhibitionDeviceGroupsApi, AbstractApi()
                     sourceDeviceGroup = sourceDeviceGroup,
                     targetRoom = sourceDeviceGroup.room
                         ?: return createBadRequest("Source device group $sourceDeviceGroupId has no room"),
-                    creatorId = userId
+                    creatorId = userId,
+                    targetContentVersionExhibition = null
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/noheva/api/ExhibitionDevicesGroupsApiImpl.kt:93]: You should not be able to change targetContentVersionExhibition, the target exhibition needs to be derived from the targetRoom (as targetExhibition)

File: src/main/kotlin/fi/metatavu/noheva/devices/ExhibitionDeviceGroupController.kt [hunk 2]
```diff
@@ -147,12 +143,14 @@ class ExhibitionDeviceGroupController {
...
    *
    * @param sourceDeviceGroup source device group
+   * @param targetContentVersionExhibition if not empty then new content versions should be assigned to this exhibition
    * @param creatorId id of user that created the copy
    * @return copied device group
    */
   fun copyDeviceGroup(
     idMapper: IdMapper,
     sourceDeviceGroup: ExhibitionDeviceGroup,
+    targetContentVersionExhibition: Exhibition?,
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/noheva/devices/ExhibitionDeviceGroupController.kt:153]: remove this, targetExhibition should be derived from targetRoom (as target exhibition)

File: src/main/kotlin/fi/metatavu/noheva/devices/ExhibitionDeviceGroupController.kt [hunk 3]
```diff
@@ -184,6 +182,7 @@ class ExhibitionDeviceGroupController {
       sourceDeviceGroup = sourceDeviceGroup,
       targetDeviceGroup = targetDeviceGroup,
       idMapper = idMapper,
+      targetContentVersionExhibition = targetContentVersionExhibition,
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/noheva/devices/ExhibitionDeviceGroupController.kt:185]: remove this, targetExhibition should be derived from targetDeviceGroup

File: src/main/kotlin/fi/metatavu/noheva/exhibitions/ExhibitionController.kt [hunk 4]
```diff
@@ -231,14 +234,16 @@ class ExhibitionController {
...
      * @param idMapper id mapper
      * @param sourceDeviceGroups source device groups
+     * @param targetContentVersionExhibition target content version exhibition
      * @param creatorId creating user id
      * @return copied device groups
      */
     private fun copyDeviceGroups(
         idMapper: IdMapper,
         sourceDeviceGroups: List<ExhibitionDeviceGroup>,
+        targetContentVersionExhibition: Exhibition,
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/noheva/exhibitions/ExhibitionController.kt:244]: replace with targetExhibition

File: src/main/kotlin/fi/metatavu/noheva/exhibitions/ExhibitionController.kt [hunk 5]
```diff
@@ -247,6 +252,7 @@ class ExhibitionController {
                 idMapper = idMapper,
                 sourceDeviceGroup = sourceDeviceGroup,
                 targetRoom = targetRoom,
+                targetContentVersionExhibition = targetContentVersionExhibition,
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/noheva/exhibitions/ExhibitionController.kt:255]: replace with targetExhibition

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: src/main/kotlin/fi/metatavu/noheva/devices/ExhibitionDeviceGroupController.kt [hunk 6]
```diff
@@ -338,37 +330,35 @@ class ExhibitionDeviceGroupController {
   }
 
   /**
-   * Copies group content versions
+   * Copies content versions from source device group to target device group
    *
-   * @param sourceGroupContentVersions copy source group content versions
+   * @param contentVersions copy source group content versions
```
Comment (villejuutila) [inline src/main/kotlin/fi/metatavu/noheva/devices/ExhibitionDeviceGroupController.kt:335]:
```suggestion
   * @param contentVersions copy source content versions
```

File: src/main/kotlin/fi/metatavu/noheva/exhibitions/ExhibitionController.kt [hunk 7]
```diff
@@ -231,14 +234,16 @@ class ExhibitionController {
...
      * @param creatorId creating user id
      * @return copied device groups
      */
     private fun copyDeviceGroups(
         idMapper: IdMapper,
         sourceDeviceGroups: List<ExhibitionDeviceGroup>,
+        targetContentVersionExhibition: Exhibition,
         creatorId: UUID
-    ): List<ExhibitionDeviceGroup> {
+        ): List<ExhibitionDeviceGroup> {
```
Comment (villejuutila) [inline src/main/kotlin/fi/metatavu/noheva/exhibitions/ExhibitionController.kt:246]:
```suggestion
      ): List<ExhibitionDeviceGroup> {
```

**Review by villejuutila — CHANGES_REQUESTED** [review]
Requires some more investigation after these two reviews.

File: src/main/kotlin/fi/metatavu/noheva/devices/ExhibitionDeviceGroupController.kt [hunk 8]
```diff
@@ -282,18 +273,18 @@ class ExhibitionDeviceGroupController {
...
+      exhibitionRoom = null
     )
 
     // Assign ids for target resources
 
     sourceDevices.map(ExhibitionDevice::id).map(idMapper::assignId)
     sourceAntennas.map(RfidAntenna::id).map(idMapper::assignId)
     sourcePages.map(ExhibitionPage::id).map(idMapper::assignId)
-    sourceGroupContentVersions.map(GroupContentVersion::id).map(idMapper::assignId)
+    sourceGroupContentVersions.map(ContentVersion::id).map(idMapper::assignId)
```
[the remark this answers is not in the document]
Comment (kaeline) [reply src/main/kotlin/fi/metatavu/noheva/devices/ExhibitionDeviceGroupController.kt:287]: fixed the name

File: src/main/kotlin/fi/metatavu/noheva/api/ExhibitionDevicesGroupsApiImpl.kt [hunk 9]
```diff
@@ -22,6 +22,7 @@ import javax.ws.rs.core.Response
  */
 @RequestScoped
 @Transactional
+@Suppress ("unused")
```
Comment (villejuutila) [inline src/main/kotlin/fi/metatavu/noheva/api/ExhibitionDevicesGroupsApiImpl.kt:25]:
```suggestion
@Suppress("unused")
```

File: src/main/kotlin/fi/metatavu/noheva/api/ExhibitionDevicesGroupsApiImpl.kt [hunk 10]
```diff
@@ -168,10 +163,10 @@ class ExhibitionDevicesGroupsApiImpl : ExhibitionDeviceGroupsApi, AbstractApi()
         val exhibitionDeviceGroup = exhibitionDeviceGroupController.findDeviceGroupById(deviceGroupId)
             ?: return createNotFound("Room $deviceGroupId not found")
 
-        val groupContentVersions = groupContentVersionController.listGroupContentVersions(
+        val groupContentVersions = contentVersionController.listContentVersions(
```
Comment (villejuutila) [inline src/main/kotlin/fi/metatavu/noheva/api/ExhibitionDevicesGroupsApiImpl.kt:166]:
```suggestion
        val contentVersions = contentVersionController.listContentVersions(
```
For clarity

**Review by villejuutila — APPROVED** [review]

**Pull request merged**